dozer-types/src/json_types.rs

```diff
@@ -33,13 +80,115 @@ pub fn field_to_json_value(field: Field) -> Value {
         Field::Decimal(n) => Value::String(n.to_string()),
         Field::Timestamp(ts) => Value::String(ts.to_rfc3339_opts(SecondsFormat::Millis, true)),
         Field::Date(n) => Value::String(n.format(DATE_FORMAT).to_string()),
-        Field::Bson(b) => Value::from(b),
+        Field::Json(b) => json_value_to_serde_json(b).unwrap(),
         Field::Point(point) => convert_x_y_to_object(&point.0.x_y()),
         Field::Duration(d) => convert_duration_to_object(&d),
         Field::Null => Value::Null,
     }
 }
 
+pub fn json_value_to_serde_json(value: JsonValue) -> Result<Value, DeserializationError> {
+    match value {
+        JsonValue::Null => Ok(Value::Null),
+        JsonValue::Bool(b) => Ok(Value::Bool(b)),
+        JsonValue::Number(n) => {
+            if n.0.is_finite() {
+                Ok(json!(n.0))
+            } else {
+                Err(DeserializationError::F64TypeConversionError)
+            }
+        }
+        JsonValue::String(s) => Ok(Value::String(s)),
+        JsonValue::Array(a) => Ok(Value::Array(
+            a.into_iter()
+                .map(|val| json_value_to_serde_json(val).unwrap())
```

Comment:
Please don't `unwrap`. We make make use of the fact that `Result<Vec<T>, Error>` can be a `collect` output type and write

```rust
a.into_iter().map(|value| json_value_to_serde_json(val)).collect::<Result<Vec<_>, _>>()?
```

Reply:
> an iterator over elements of type `serde_json::Value`

This is because of the `unwrap`.

Either way is fine.